src/gnssro/CMakeLists.txt

```diff
@@ -7,6 +7,9 @@ if( HAVE_BUFRLIB_AND_NETCDF )
     set(GNSSRO_SRC gnssro_bufr2ioda.f90)
     add_executable(gnssro_bufr2ioda ${GNSSRO_SRC})
     target_link_libraries(gnssro_bufr2ioda ${bufrlib_LIBRARIES})
+    if(CMAKE_Fortran_COMPILER_ID MATCHES GNU AND CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
```

Comment:
Does this have the issue where if you set FC=mpifort, and gfortran is the underlying compiler, this doesn't match (because "mpifort" is returned instead of "GNU") which results in skipping the following target_compile_options() command when you wanted to include that command.

Reply:
Very good question, but I think it will continue to work correctly.  I can use this test script which jedi-rapids uses to determine the compiler as an example of why it works:

On my home system:
```
mjo@wwwyzzerdd $ cmake -DTMPDIR=/tmp -DLANG=Fortran -P ~/.jedi-rapids/repository/src/jedirapids/scripts/get_cmake_lang_compiler.cmake 
-- The Fortran compiler identification is GNU 10.1.0
CMAKE_Fortran_COMPILER_ID: GNU
CMAKE_Fortran_COMPILER_VERSION: 10.1.0
mjo@wwwyzzerdd $ FC=mpifort cmake -DTMPDIR=/tmp -DLANG=Fortran -P ~/.jedi-rapids/repository/src/jedirapids/scripts/get_cmake_lang_compiler.cmake 
-- The Fortran compiler identification is GNU 10.1.0
CMAKE_Fortran_COMPILER_ID: GNU
CMAKE_Fortran_COMPILER_VERSION: 10.1.0
mjo@wwwyzzerdd $ mpifort --version
GNU Fortran (Gentoo 10.1.0-r1 p2) 10.1.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

On s4-submit:
```
molah@s4-submit ~ $ cmake -DTMPDIR=/tmp -DLANG=Fortran -P ~/.jedi-rapids/repository/src/jedirapids/scripts/get_cmake_lang_compiler.cmake
-- The Fortran compiler identification is Intel 17.0.6.20171215
CMAKE_Fortran_COMPILER_ID: Intel
CMAKE_Fortran_COMPILER_VERSION: 17.0.6.20171215
molah@s4-submit ~ $ echo $FC
mpiifort
molah@s4-submit ~ $ which $FC
/opt/intel/17/compilers_and_libraries_2017.6.256/linux/mpi/intel64/bin/mpiifort
```

Let me know if that is not the case for your system.

Reply:
@markjolah, the cmake commands you show above work correctly on my iMac. I've currently got GNU 9.2.0 installed.

```
sysadmins-iMac:jedi-rapids stephenh$ cmake -DTMPDIR=/tmp -DLANG=Fortran -P ~/.jedi-rapids/repository/src/jedirapids/scripts/get_cmake_lang_compiler.cmake
-- The Fortran compiler identification is GNU 9.2.0
CMAKE_Fortran_COMPILER_ID: GNU
CMAKE_Fortran_COMPILER_VERSION: 9.2.0

sysadmins-iMac:jedi-rapids stephenh$ FC=mpifort cmake -DTMPDIR=/tmp -DLANG=Fortran -P ~/.jedi-rapids/repository/src/jedirapids/scripts/get_cmake_lang_compiler.cmake
-- The Fortran compiler identification is GNU 9.2.0
CMAKE_Fortran_COMPILER_ID: GNU
CMAKE_Fortran_COMPILER_VERSION: 9.2.0
```

Plus, this feature branch works on my iMac as well.